### ISSUE TITLE
Chibi-scheme: Crashes on FFI pointer write in FreeBSD amd64

### ISSUE THREAD

**Issue by okuoku** [okuoku/yuni#37]
Perhaps `integer->ptr` does not return valid `Cpointer` object. Overwrapping heap?

```scheme
  (define (dlopen path)
    (define namebuf (malloc 4096))
    (ptr-write/asciiz! (integer->ptr namebuf) 0 4096 path)
    (let ((p (call-bootstrap YUNIBOOTSTRAP0_DLOPEN namebuf 0 0)))
     (free namebuf)
     (let ((err (car p))
           (ptr (cdr p)))
       (and (= err 0)
            ptr))))
```

```
(gdb) bt
#0  0x000000080060dcef in dlopen_object (
    name=0x801446000 '▒' <repeats 200 times>..., fd=-1, refobj=0x800627000,
    lo_flags=2, mode=2, lockstate=<value optimized out>)
    at /usr/src/libexec/rtld-elf/rtld.c:3101
#1  0x000000080060b1d1 in rtld_dlopen (
    name=0x801446000 '▒' <repeats 200 times>..., fd=-1,
    mode=<value optimized out>) at /usr/src/libexec/rtld-elf/rtld.c:3069
#2  0x000000080342529c in do_dlopen (ptr_name=34380996608,
    out_err=0x801ab2250, out_ptr=0x801ab2258) at bootstrap.inc.c:28
#3  0x00000008034254ab in yuniffi_bootstrap0 (in=0x801ab2210, in_len=4,
    out=0x801ab2250, out_len=2) at bootstrap.inc.c:64
#4  0x00000008034253ee in yuniffi_nccc_call (func=0x803425420, in=0x801ab2200,
    in_offset=0, in_len=4, out=0x801ab2240, out_offset=0, out_len=2)
    at /yuniroot/yuni/yunistub/chibi/yuniffi_stub.c:19
#5  0x0000000803427bd0 in sexp_yuniffi_nccc_call_stub (ctx=0x801600060,
    self=0x801c91840, n=7, arg0=0x801ce1780, arg1=0x801ab2200, arg2=0x1,
    arg3=0x11, arg4=0x801ab2240, arg5=0x1, arg6=0x9)
    at /yuniroot/build/yuni/yunistub/chibi/yuniffi.stub.c:170
#6  0x000000080084e01b in sexp_fcall (ctx=0x1, self=<value optimized out>,
    n=<value optimized out>, f=<value optimized out>) at fcall.c:21
#7  0x000000080084ea66 in sexp_apply (ctx=0x801600060,
    proc=<value optimized out>, args=<value optimized out>) at vm.c:1292
#8  0x0000000000404275 in run_main (argc=10, argv=0x7fffffffea00) at main.c:670
#9  0x0000000000404eab in main (argc=1, argv=0x800824888) at main.c:721
```

**Comment by okuoku** [comment]
Disabled test: https://github.com/okuoku/yuni/commit/01dca0521a3038e26f04cabc46d42ab977d89712

**Comment by okuoku** [comment]
Fixed by #47. CLOSE.

Test enabled back at https://github.com/okuoku/yuni/commit/c0a960c44f2a7b31779200d6f3bc38c1b5a549f2
